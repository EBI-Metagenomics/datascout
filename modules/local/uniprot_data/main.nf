process UNIPROT_DATA {

    conda "./environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/d2/d2cc550ff67f8541d44dc2db1b5d2d2e1cfccfe8536222b49788deefde7460f0/data' :
        'community.wave.seqera.io/library/python_pip_biopython_requests:725bda83fb97ec48' }"

    debug true
    publishDir "${params.output}", mode: "copy"
    label "process_low"

    tag "${meta}"

    errorStrategy 'retry'
    maxRetries 2

    input:
      tuple val(meta), path(tax_ranks), val(rank), val(evidence)

    output:
      tuple val(meta), path("${meta.id}_uniprot_dir"), emit: uniprot_results
      path "versions.yml", emit: versions

    script:
    """
    mkdir -p ${meta.id}_uniprot_dir
    uniprot_data.py --tax_file ${tax_ranks} --output "${meta.id}_uniprot_dir" --rank ${rank} --evidence ${evidence}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        \$( uniprot_data.py --tax_file ${tax_ranks} --output "${meta.id}_uniprot_dir" --version 2>&1 )
    END_VERSIONS
    """
}

// Get proteins from uniprot and reformat headers
