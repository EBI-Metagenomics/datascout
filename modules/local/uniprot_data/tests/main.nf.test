nextflow_process {

    name "Test Process UNIPROT_DATA"
    script "modules/local/uniprot_data/main.nf"
    process "UNIPROT_DATA"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = tuple([id:"test_sample1", genome_id:"GCA_003719455.1", ena_tax:"default"], "${projectDir}/tests/reference_databases/test_sample1_tax_ranks.tsv", "default", 2)
                input[1] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.uniprot_results,
                process.out.versions
            ).match()
        }

    }

}
