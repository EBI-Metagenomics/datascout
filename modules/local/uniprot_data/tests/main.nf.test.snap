{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_sample1",
                        "genome_id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/543b1f98aff365c1bc545a58cc4bbf72/work/71/10f35861d5e0f2168739b0578d82bc/test_sample1_uniprot_dir"
                ]
            ],
            [
                "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/543b1f98aff365c1bc545a58cc4bbf72/work/71/10f35861d5e0f2168739b0578d82bc/versions.yml"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-09T09:39:33.762252"
    }
}